### ISSUE TITLE
Reports with relative links, scripts, and images?

### ISSUE THREAD

**Issue by TheXenocide** [blakyaks/azure-pipeline-html-reports#9]
Hello. I am trying to understand if this extension will serve my needs or if it can at least serve as a good foundation if I need to create my own extension (very new to this space though, definitely not familiar with making these extensions). After an initial review of the source here, I suspect it doesn't quite serve my needs exactly as is, but maybe it can be modified (either here or in my own fork) to accomplish my goals?

Essentially, I have a CLI tool that produces a report directory during one of my build steps. There are nested directories and child reports which are all linked to using relative URLs. I'm hoping to generate a single report tab which points to the root index.html file which would then allow the user to click on links within that report which point to other HTML files, viewers, and file downloads. These HTML files use css and JavaScript files and the SVG Viewer utility loads images, all of which are specified using URLs relative to the HTML file using/linking to them.

Here is an example of the directory structure:

```
┬Reports
├┬Comprehensive
│├┬PackageDependencies
││├─Microsoft.Bcl.AsyncInterfaces.dgml
││├─Microsoft.Bcl.AsyncInterfaces.svg
││├─...
││├─System.Threading.Tasks.Extensions.dgml
││└─System.Threading.Tasks.Extensions.svg
│├┬PackageRelationships
││├─Microsoft.Bcl.AsyncInterfaces.dgml
││├─Microsoft.Bcl.AsyncInterfaces.svg
││├─...
││├─System.Threading.Tasks.Extensions.dgml
││└─System.Threading.Tasks.Extensions.svg
│├┬RestoredVersions
││├─Microsoft.Bcl.AsyncInterfaces.dgml
││├─Microsoft.Bcl.AsyncInterfaces.svg
││├─...
││├─System.Threading.Tasks.Extensions.dgml
││└─System.Threading.Tasks.Extensions.svg
│├─Comprehensive.PackageRelationships.html
│├─Comprehensive.ProjectRelationshipsGraph.dgml
│├─Comprehensive.ProjectRelationshipsGraph.svg
│├─Comprehensive.RestoredPackages.html
│├─Comprehensive.Statistics.html
│├─Comprehensive.TransitiveDependencyGraph.dgml
│├─Comprehensive.TransitiveDependencyGraph.svg
│├─dependency-report.css
│├─index.html
│├─svg-viewer.js
│└─SVGViewer.html
├┬Solution
│├─CachingPackageTest.PackageRelationships.html
│├─CachingPackageTest.ProjectRelationshipsGraph.dgml
│├─CachingPackageTest.ProjectRelationshipsGraph.svg
│├─CachingPackageTest.RestoredPackages.html
│├─CachingPackageTest.Statistics.html
│├─CachingPackageTest.TransitiveDependencyGraph.dgml
│├─CachingPackageTest.TransitiveDependencyGraph.svg
│├─dependency-report.css
│├─index.html
│├─svg-viewer.js
│└─SVGViewer.html
├─dependency-report.css
├─index.html
├─svg-viewer.js
└─SVGViewer.html
```

From what I can see in the source, it looks like maybe the current implementation only gathers html files using file globbing, so I suspect all my css/js/dgml/svg files will not be included. Additionally, it seems that it will generate its own links to each HTML file, though I really only need it to point to Reports/index.html and allow the links there to be followed to the various other html files.

Is there any possibility of adding support for that here (I would be willing to submit a PR, though I might need some guidance) or, if not, could you possibly help me understand what types of changes I would need to look at making if I forked this repo? It seems sooo close to what I'm looking for.

Thanks kindly for the help.

**Comment by TheXenocide** [comment]
I think I'm starting to understand the source a little better, but it's not very promising. I guess the only way to show the content from an HTML report is to download it from an attachment via API calls and then render it into an iframe manually? There's no URL for directly serving the html/css/etc. from inside attachments/artifacts then? It's a shame because we're using other CI environments as well and it's much easier to embed externally generated reports than this. Any ideas besides writing a completely custom report generator for this?

**Comment by blakyaks** [comment]
Hey @TheXenocide 

Sorry for the delay, the issues you're getting are a limitation of the extension really - it was designed for simple single page HTML reports and doesn't act as a web server since the pages are rendered within ADO based on backend artifacts. To change this would be basically a new extension and I'm not even sure it would be possible within the ADO extension framework, so not something we'd be looking to implement on given the effort involved.

I've added a note on the limitations on the [Marketplace](https://marketplace.visualstudio.com/items?itemName=blakyaks.azure-pipelines-html-reports) as we've had a similar issue reported - at the least this makes it clear what the extension use case is.

**Comment by TheXenocide** [comment]
Yeah, thanks kindly for following up in either case; we've also come to the conclusion that we will need to make a custom extension if we want this support given we'll probably need to make the HTML/scripts/etc. capable of loading from task attachments directly by API as it does not appear there's any way to serve up the raw HTML/JS/CSS with relative paths or anything like that without hosting a separate server. I think this extension repo will at least serve as a decent starting point even if much of the meat and potatoes will be replaced, so thanks for the foundation at least :)
